crates/rooch-types/src/cli.rs

```diff
@@ -61,3 +68,17 @@ impl From<io::Error> for CliError {
         CliError::IOError(e.to_string())
     }
 }
+
+#[async_trait]
+pub trait CommandAction<T: Serialize + Send>: Sized + Send {
+    /// Executes the command, returning a command specific type
+    async fn execute(self) -> CliResult<T>;
+
+    /// Executes the command, and serializes it to the common JSON output type
+    async fn execute_serialized(self) -> CliResult<String> {
+        match self.execute().await {
+            Ok(result) => Ok(serde_json::to_string_pretty(&result).unwrap()),
+            Err(e) => Err(e),
```

Comment:
Make the error also a json? For convenience parse the cli output.

Reply:
Command with error will exit with code 1. And the error is a enum type will print a single `String` output.

You mean define a `struct` type for error like:
```
struct CliError {
  code: int,
  kind: ErrorKind,
  message: String,
```

Reply:
Or wrap the whole result? like

```
{"ok": {xxx}}
{"error": "message"}
```

I merge this PR first, let this TBD.